_posts/2019-05-31-ucx-dgx.md

```diff
@@ -0,0 +1,531 @@
+---
+layout: widepost
+title: Experiments in High Performance Networking with UCX and DGX
+author: Matthew Rocklin
+draft: true
+tags: [python,scipy]
+theme: twitter
+---
+{% include JB/setup %}
+
+*This post is about experimental and rapidly changing software.
+Code examples in this post should not be relied upon to work in the future.*
+
+Executive Summary
+-----------------
+
+This post talks about connecting UCX, a high performance networking library, to
+Dask, a parallel Python library, to accelerate communication-heavy workloads,
+particularly when using GPUs.
+
+Additionally, we do this work on a DGX, a high-end multi-CPU multi-GPU machine
+with a complex internal network.  Working in this context was good to force
+improvements in setting up Dask in heterogeneous situations targetting
+different network cards, CPU sockets, GPUs, and so on..
+
+Moviation
+---------
+
+Many distributed computing workloads are communication-bound.
+This is common in cases like the following:
+
+1.  Dataframe joins
+2.  Machine learning algorithms
+3.  Rearranging arrays
+4.  ...
+
+Communication also increasingly becomes a bottlenck when our computation
+becomes faster, such as increasingly the case when we use GPUs for computing.
+
+Historically high performance communication was only commonly available using
+MPI, or with very custom codes.  We were trying to get close to the
```

Comment:
I incorrectly read "We were trying" as "We tried and gave up...". Maybe something like "Our goal is to approach the high communicate performance of MPI, while retaining the programmability and accessibility of a dynamic system like Dask".